File: TreasurersApp/Models/PhoneTypeActionResult.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace TreasurersApp.Models
{
    public class PhoneTypeActionResult : WebResult<PhoneType>
    {
        public PhoneTypeActionResult(bool success, List<string> statusMessages, PhoneType phoneType)
            :base(success, statusMessages, phoneType)
        {
        }
    }
}

File: TreasurersApp/ClientApp/src/app/models/PhoneTypeActionResult.ts
  import { PhoneType } from './PhoneType';
  import { PhoneTypeService } from '../maintenance/phone-type-maintenance/phone-type.service';

  export class PhoneTypeActionResult {
    success = false;
    statusMessages: string[] = [];
    phoneType: PhoneType = null;

    constructor(private phoneTypeService: PhoneTypeService) {
      this.success = false;
      this.phoneType = this.phoneTypeService.newPhoneType();
      this.statusMessages = [];
    }
  }
